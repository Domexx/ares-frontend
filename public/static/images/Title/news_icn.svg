<svg width="22" height="36" viewBox="0 0 22 36" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<rect width="22" height="36" fill="url(#pattern0)"/>
<defs>
<pattern id="pattern0" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image0_285_315" transform="scale(0.0454545 0.0277778)"/>
</pattern>
<image id="image0_285_315" width="22" height="36" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABYAAAAkCAYAAACNBsqdAAABfElEQVRIS7WXsXXDMAxEyZmyQTqX7jKBNJM0QVLZnbtskJmYd4pOD4QJCJQcNn6SzLuPIyjTObVHWW9n4/nubT1xESylpPvXPV0/rhToNuCESvDx/VgEp2k6bADhAkKMnP98SHzGoBLe6n6BwSY8juNSuhxnKqiIIc5szxo0o2jR965BFQUpGcmZCp6ISaupo2sAKKyNuXgtAy8itCieX94vy8YKLZ420RGB8PZ5S+j7J2LZBa1s9T15DWGX+Cfn9LbuQhhZBqCSsYSItbhlwPss3SXGlymMT4y9CtjfuxlTUIt6axDKmII9ebsZz/OchmHYwCxhvTsZhZsxxDFoIMXZIdXrb/0xCBNrA9kZlnAXsTT4N2KSaoNQH+sI5LV+pre02ceyK1oZk7j1bLePe4jld0PvCpC9nFi3krz2qrGIMX87CXniVjVWxlLLNDhKrEHDBt67wqs+FJGX8V60rkEk40MGJJbn6e4D9epcVcDDI07CJDsqzPnmX4pf1fpDKlZsYcEAAAAASUVORK5CYII="/>
</defs>
</svg>
