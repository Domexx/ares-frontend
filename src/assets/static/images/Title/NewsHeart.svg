<svg width="24" height="36" viewBox="0 0 24 36" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<rect width="22" height="36" fill="url(#pattern0)"/>
<rect x="11" y="4" width="13" height="12" fill="url(#pattern1)"/>
<defs>
<pattern id="pattern0" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image0_655_2" transform="scale(0.0454545 0.0277778)"/>
</pattern>
<pattern id="pattern1" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image1_655_2" transform="scale(0.0769231 0.0833333)"/>
</pattern>
<image id="image0_655_2" width="22" height="36" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABYAAAAkCAYAAACNBsqdAAABfElEQVRIS7WXsXXDMAxEyZmyQTqX7jKBNJM0QVLZnbtskJmYd4pOD4QJCJQcNn6SzLuPIyjTObVHWW9n4/nubT1xESylpPvXPV0/rhToNuCESvDx/VgEp2k6bADhAkKMnP98SHzGoBLe6n6BwSY8juNSuhxnKqiIIc5szxo0o2jR965BFQUpGcmZCp6ISaupo2sAKKyNuXgtAy8itCieX94vy8YKLZ420RGB8PZ5S+j7J2LZBa1s9T15DWGX+Cfn9LbuQhhZBqCSsYSItbhlwPss3SXGlymMT4y9CtjfuxlTUIt6axDKmII9ebsZz/OchmHYwCxhvTsZhZsxxDFoIMXZIdXrb/0xCBNrA9kZlnAXsTT4N2KSaoNQH+sI5LV+pre02ceyK1oZk7j1bLePe4jld0PvCpC9nFi3krz2qrGIMX87CXniVjVWxlLLNDhKrEHDBt67wqs+FJGX8V60rkEk40MGJJbn6e4D9epcVcDDI07CJDsqzPnmX4pf1fpDKlZsYcEAAAAASUVORK5CYII="/>
<image id="image1_655_2" width="13" height="12" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA0AAAAMCAYAAAC5tzfZAAAAV0lEQVR42mNgQID/SJiBCHGG/4/sTf//r24CYyQFuMRRJUAYxMcnjqEJxEbG6OIomojFFGkyJlYjVIMxLDAIakTXQFAjLg04NRLSgKIRCRPUgKzRGJcGAL6b08s4nijiAAAAAElFTkSuQmCC"/>
</defs>
</svg>
