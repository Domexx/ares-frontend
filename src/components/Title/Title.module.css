/* === === === === Title === === === === */

.Title {
  display: flex;
  flex-direction: row;
}

.Title-container {
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
}

.Title-title {
  font-size: 14px;
  font-weight: 800;
  text-transform: uppercase;
}

.Title-description {
  font-size: 12px;
}
