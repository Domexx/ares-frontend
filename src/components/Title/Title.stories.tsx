import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import TitleComponent, { Props } from './Title';

const Component = {
  title: 'Elements/Title',
  component: TitleComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof TitleComponent>> = (args) => {
  return <TitleComponent {...args} />;
};

export const Title = Template.bind({});
export default Component;

// https://storybook.js.org/docs/react/essentials/controls
Title.args = {
  title: 'Ares lmao',
  description: 'This is a very cool Title',
  iconIdentifier: 'news',
} as Props;
