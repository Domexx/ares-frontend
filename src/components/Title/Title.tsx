import React, { PropsWithChildren } from 'react';
import s from './Title.module.css';

import { ReactComponent as News } from '../../assets/static/images/Title/News.svg';
import { ReactComponent as NewsWithHeart } from '../../assets/static/images/Title/NewsHeart.svg';

export type Props = {
  title: string;
  description: string;
  iconIdentifier: string;
};

const Title: React.FC<PropsWithChildren<Props>> = ({ title, description, iconIdentifier, children }) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const icons: any = {
    news: <News />,
    newswithHeart: <NewsWithHeart />,
  };

  const selectedIcon = icons[iconIdentifier];

  return (
    <div className={s['Title']}>
      {selectedIcon}
      <div className={`${s['Title-container']}`}>
        <h2 className={`${s['Title-name']}`}>{title}</h2>
        <p className={`${s['Title-description']}`}>{description}</p>
      </div>
      {children}
    </div>
  );
};

export default Title;
