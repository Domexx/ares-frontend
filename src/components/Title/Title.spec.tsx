import React from 'react';

import { render } from '@testing-library/react';

import Title from './Title';

describe('Title Component', () => {
  it('should render properly', () => {
    const { container } = render(<Title title="Test Title" description="Test description" iconIdentifier="news" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
