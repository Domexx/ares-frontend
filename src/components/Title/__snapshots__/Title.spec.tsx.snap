// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component should render properly 1`] = `
<div
  class="Title"
>
  <svg />
  <div
    class="Title-container"
  >
    <div
      class="Title-title"
    >
      Test Title
    </div>
    <div
      class="Title-description"
    >
      Test description
    </div>
  </div>
</div>
`;
